src/room007/models/model.py

```diff
@@ -38,6 +44,43 @@ def option_combination_name(options):
         else:
             return [('||| no parameter combinations |||', {option_name: option.default for option_name, option in self.options.items()})]
 
+
+class FeatureManager(object):
+    save = False
```

Comment:
For second question: Because then each decorated function need to know about the inner working of the decorating function: decorator(function(args)) -> function(args) need to be independent of caller

For first question:
Decorators are executed on start time, before we assign any state to the 'save' variable.
So the moment we want to set the 'save' to true all instances are already instantiated, and because we don't have any pointer to those instances, we can't modify an instance state. What we can do, is change the class variables. 
(or to say differently, this is a group behavior, not something that will be different from instance to instance0